nameserver/mdns_client.go

```diff
@@ -64,17 +84,21 @@ type mDNSQueryInfo struct {
 }
 
 func NewMDNSClient() (*MDNSClient, error) {
-	if conn, err := net.ListenUDP("udp4", &net.UDPAddr{IP: net.IPv4zero, Port: 0}); err != nil {
```

Comment:
It did not make sense to listen for anything here. Binding to ports should be delayed until we `Start()` the server...